### PR TITLE
Fix verify share table items and access point share no bucket policy

### PR DESCRIPTION
**Pull request by noah-paige**
### Feature or Bugfix

- Bugfix

### Detail
- Verify Share throws errors on verifying table shares since introducing the new way to `initialize_clients()` for LF Shares when determining the source Glue DB
  - This PR includes the `initialize_client()` call before starting to verify the table share item
  - This PR fixes verify share checks on table shares where the glue DB is hosted in a separate catalog account

- When processing a folder item share - it will fail if there is no pre-existing bucket policy on the S3 bucket
  - This PR adds checks similar to how we handle with bucket sharing method `get_bucket_policy_or_default()`

### Relates
- N/A

### Security
Please answer the questions below briefly where applicable, or write `N/A`. Based on
[OWASP 10](https://owasp.org/Top10/en/).

- Does this PR introduce or modify any input fields or queries - this includes
fetching data from storage outside the application (e.g. a database, an S3 bucket)?
  - Is the input sanitized?
  - What precautions are you taking before deserializing the data you consume?
  - Is injection prevented by parametrizing queries?
  - Have you ensured no `eval` or similar functions are used?
- Does this PR introduce any functionality or component that requires authorization?
  - How have you ensured it respects the existing AuthN/AuthZ mechanisms?
  - Are you logging failed auth attempts?
- Are you using or adding any cryptographic features?
  - Do you use a standard proven implementations?
  - Are the used keys controlled by the customer? Where are they stored?
- Are you introducing any new policies/roles/users?
  - Have you used the least-privilege principle? How?

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by noah-paige** [comment]
Tested this PR for the following:

- Create Share for only Folder Share Item on Imported Dataset with no S3 Bucket Policy
- [x] Bucket Policy created correctly
- [x] Share Succeeds
- [x] Can Access Folder as Requestor Principal IAM Role


- Start Verify Share Items on Bucket and Table Items with Glue DB hosted in a separate Catalog Account in Data.all
- [x] Share Verify successfully runs and determines share remains in healthy status (as expected)

**Comment by dlpzx** [comment]
Thanks for the PR @noah-paige. I confirm both errors and I am testing it. I will approve and merge once it is reviewed and tested

File: backend/dataall/modules/dataset_sharing/services/share_managers/s3_access_point_share_manager.py [hunk 1]
```diff
@@ -127,10 +127,54 @@ def manage_bucket_policy(self):
         logger.info(f"Manage Bucket policy for {self.bucket_name}")
 
         s3_client = S3Client(self.source_account_id, self.source_environment.region)
```
Comment (dlpzx) [inline backend/dataall/modules/dataset_sharing/services/share_managers/s3_access_point_share_manager.py:129]: Minor: this client is never used, it is duplicated in line 154

**Review by dlpzx — APPROVED** [review]
tested!

File: backend/dataall/modules/dataset_sharing/services/share_managers/s3_access_point_share_manager.py (hunk 1 above)
Comment (noah-paige) [reply backend/dataall/modules/dataset_sharing/services/share_managers/s3_access_point_share_manager.py]: good catch - removed

**Pull request merged**